### PR TITLE
fix: enable to release selection after releasing action

### PR DESCRIPTION
**Pull request by hiroppy**
Thank you for supporting the mouse action! I'm really happy this action for me:) btw, I think the main branch doesn't support to dispose selection so this PR support it(https://www.w3.org/TR/selection-api/).  I've checked this behavior on my iPad.

main

https://user-images.githubusercontent.com/3367801/184539797-316744a9-b586-463d-8f34-7b2b3d3f5a38.MP4

this pr

https://user-images.githubusercontent.com/3367801/184539808-1245b559-a8fb-48a1-90c9-931f7edef4fe.MP4

### CODE REVIEW

**Comment by hiroppy** [comment]
oh, sorry, I found a bug that occurs when focusing input... today is late so I'll fix it tomorrow

File: Web/packages/content/src/hooks/useGestureRecognizer.ts [hunk 1]
```diff
@@ -81,7 +81,8 @@ export function useGestureRecognizer(
     const g = getClosestGesture();
     onRelease(g);
     patternConstructor.current.clear();
-  }, [onRelease, getClosestGesture]);
+    document?.getSelection()?.removeAllRanges();
```
Comment (shumbo) [inline Web/packages/content/src/hooks/useGestureRecognizer.ts:84]: Before you clear selections, could you check `g` is not null?

`releaseHandler` is called on `touchend` and `mouseup`, which means the current code would clear selections even if the gesture does not fire.

https://user-images.githubusercontent.com/10496155/184558213-d7481689-1565-467b-a82c-a9b5224953c9.mov

I believe the correct behavior is to clear selections only when a gesture is recognized. `getClosestGesture` returns `null` if there is no registered gesture that satisfies the pattern similarity threshold, so we do not want to clear selections in that case.

I'm not sure what other things we may add to this function. Putting `removeAllRanges()` inside an IF statement or using the early return pattern should both work fine (at least for now).

File: Web/packages/content/src/hooks/useGestureRecognizer.ts [hunk 2]
```diff
@@ -81,7 +81,8 @@ export function useGestureRecognizer(
...
+  }, [patternConstructor.current]);
```
Comment (shumbo) [inline Web/packages/content/src/hooks/useGestureRecognizer.ts:85]: It looks like you deleted `onRelease, getClosestGesture` and included `patternConstructor.current` to the dependencies array. Is this intentional?

There was a version mismatch, and eslint is not working correctly, but `eslint-plugin-react-hooks` shows the following warning:

Screen Shot 2022-08-15 at 8 14 15

**Review by shumbo — CHANGES_REQUESTED** [review]
@hiroppy Thanks so much for opening the PR! I really like the idea of clearing selections once an action is fired. Just making a couple of minor suggestions.

File: Web/packages/content/src/hooks/useGestureRecognizer.ts (hunk 2 above)
Comment (hiroppy) [reply Web/packages/content/src/hooks/useGestureRecognizer.ts:85]: oh, sorry, its mistake... (I don't have intention

**Comment by hiroppy** [comment]
@shumbo thank you for your review, I fixed them and it works on my device. PTAL

**Review by shumbo — APPROVED** [review]
@hiroppy Thanks for your change. This looks good to me.

You mentioned the bug with inputs (https://github.com/shumbo/Svadilfari/pull/50#issuecomment-1214389435). Is this now fixed with the non-null check? If there's no further problem, I'm willing to merge this one.

**Comment by hiroppy** [comment]
@shumbo Yep, the issue was resolved by your suggestion, thank you

**Pull request merged**